### DISCUSSION TITLE
copy app from TestFlight to App Store

### DISCUSSION THREAD

**Discussion by mvolkmann** [fastlane/fastlane#21069] [Q&A]
I have successfully used `pilot` to upload my app to TestFlight.
When beta testing is complete, can I use fastlane to promote the version of the app in TestFlight to the App Store
OR
do I have to use `deliver` to upload it again, this time from my computer to the App Store?

**Comment by EmDee** [comment]
@mvolkmann You have to call `deliver` again, but skip the upload of the `IPA`. Something along these lines:

```ruby
deliver(
  app_identifier: APP_IDENTIFIER,
  app_version: version_number,
  build_number: build_number,
  skip_binary_upload: true,
  submit_for_review: true,
  ...
)
```

**Reply by mvolkmann** [reply]
Thanks so much! I think `skip-binary-upload` was the piece I was missing.
